### DISCUSSION TITLE
Transform Data Attribute in a Dataset

### DISCUSSION THREAD

**Discussion by kyriakosschwarz** [pyg-team/pytorch_geometric#5344] [Q&A]
I created a `CustomDataset` based on `InMemoryDataset`. When I get the first Data object from the dataset it looks like this:

```
data0 = dataset[0]  # Get the first graph object.

data0

PairData(edge_index_a=[2, 26], x_a=[13, 9], edge_attr_a=[26, 3],
    edge_index_b=[2, 76], x_b=[35, 9], edge_attr_b=[76, 3],
    id=[1], y=[1], exp=[1, 908])
```
Lets say this dataset contains 1000 samples, then the `exp` attribute of this dataset would originate from a `[1000, 908]` matrix.

So, two questions: 

1. Is there a way to easily reconstruct the `[1000, 908]` `exp` matrix from all the `exp` `[1,908]` vectors of all the 1000 samples?
2. How can I apply any function (e.g. rescaling function) to the `exp` attribute of all the `PairData` objects and replace their current `exp` vectors?

**Comment by rusty1s** [comment]
1. The concatenation of features will always be in order. If you want to do it by hand you can easily do
```python
exp = torch.cat([data.exp for data in dataset], dim=0)
```
2. This is best done via [`transforms`](https://pytorch-geometric.readthedocs.io/en/latest/modules/transforms.html#torch_geometric.transforms.BaseTransform).
